crates/llama-cpp-bindings/src/engine.cc

```diff
@@ -77,6 +83,38 @@ std::string string_format(const std::string& format, Args ... args)
 	return std::string(buf.get(), buf.get() + size - 1); // We don't want the '\0' inside
 }
 
+float softmax(float* nums, size_t len, float_t temperature) {
```

Comment:
```suggestion
float softmax(const float* nums, size_t len, float temperature) {
```

Reply:
I can't make these const because my `softmax` function modifies the array in-place rather than allocating a new one. Would you rather I allocate one?